src/ppx/knob.cpp

```diff
@@ -0,0 +1,316 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "backends/imgui_impl_glfw.h"
+#include "ppx/knob.h"
+
+#include <cstring>
+
+// -------------------------------------------------------------------------------------------------
+// KnobType
+// -------------------------------------------------------------------------------------------------
+
+std::ostream& operator<<(std::ostream& os, KnobType kt)
+{
+    return os << KnobType2Str(kt);
+}
+
+std::string KnobType2Str(KnobType kt)
```

Comment:
The only use for this function is in error message in switch statement, since type enum/switch does not need to exist, this probably also don't need to exist.